### PR TITLE
test(lightpush): Lightpush functional tests

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
# Description
Implement lightpush's functional tests

# Changes
- [X] Move existing tests to its own module.
- [X] Implement functional tests.
- [x] Remove duplicated tests.
- [X] Fix some imports.
    - One of these made `waku_store/test_client` to be included in the binary, which had been forgotten, making necessary a small update to the file.

### CODE REVIEW

File: tests/waku_lightpush/test_client.nim
Comment (AlejandroCabeza) [inline tests/waku_lightpush/test_client.nim:233]: This triggers SEGFAULT.

File: tests/node/test_wakunode_lightpush.nim
Comment (Ivansete-status) [inline tests/node/test_wakunode_lightpush.nim:97]: nitpick:

```suggestion
      assert publishResponse.isOk(), publishResponse.error
```

File: tests/node/test_wakunode_lightpush.nim
Comment (Ivansete-status) [inline tests/node/test_wakunode_lightpush.nim:93]: Nitpick. For nouns, we don't use parenthesis.
```suggestion
        echo "Publish failed: ", publishResponse.error
```

File: tests/waku_lightpush/test_client.nim
Comment (Ivansete-status) [reply tests/waku_lightpush/test_client.nim:237]: Interesting, we need to debug why this happens, although it seems some kind of overflow

File: tests/waku_lightpush/test_client.nim
Comment (Ivansete-status) [inline tests/waku_lightpush/test_client.nim:93]: Nitpick: I think assert is more convenient when dealing with `Result`. The recommendation applies elsewhere.
```suggestion
      assert publishResponse.isOk(), publishResponse.error
      check handlerFuture.finished()
```

**Review by Ivansete-status — COMMENTED** [review]
Wonderful PR! :partying_face: 
I don't approve it yet because we need to know why the seg fault is happening.
Cheers

File: tests/resources/pubsub_topics.nim
Comment (gabrielmer) [inline tests/resources/pubsub_topics.nim:2]: I think pubsub is mispelled in the proc's name.

And also believe that the `contentTopicName` parameter is kind of misnamed. Content topic is not part of the pubsub topic name

**Review by gabrielmer — COMMENTED** [review]
LGTM, thanks so much!
Same, let's first figure out the segfault before proceeding

File: tests/waku_lightpush/test_client.nim
Comment (AlejandroCabeza) [reply tests/waku_lightpush/test_client.nim:237]: Indeed, this has already been reported here: https://github.com/waku-org/nwaku/issues/2271

File: tests/node/test_wakunode_lightpush.nim
Comment (AlejandroCabeza) [reply tests/node/test_wakunode_lightpush.nim:93]: I understand nim understands this as calling the method?

File: tests/resources/pubsub_topics.nim
Comment (AlejandroCabeza) [reply tests/resources/pubsub_topics.nim:2]: Ah, yes, oversight, thanks!

File: tests/node/test_wakunode_lightpush.nim
Comment (AlejandroCabeza) [reply tests/node/test_wakunode_lightpush.nim:97]: The reason I changed to using `check` over `assert` is I got used to it because of the `SEGFAULT` when the assertion is false.
Given we found that didn't happen anymore, I was about to update it with your recommendation, but I tried it again and it still happens (at least on my machine), so... 😅
Any opinions/ideas?

**Comment by AlejandroCabeza** [comment]
Any ideas as to why the `test-ubuntu-latest` fails despite  `[Summary] 34 tests run (0.00s): 34 OK, 0 FAILED, 0 SKIPPED`?

File: tests/node/test_wakunode_lightpush.nim
Comment (Ivansete-status) [reply tests/node/test_wakunode_lightpush.nim:97]: Ok disregard my comment in this case. let's keep the one that doesn't crash :)

File: tests/node/test_wakunode_lightpush.nim
Comment (Ivansete-status) [reply tests/node/test_wakunode_lightpush.nim:93]: > I understand nim understands this as calling the method?

Correct

**Comment by Ivansete-status** [comment]
> Any ideas as to why the `test-ubuntu-latest` fails despite `[Summary] 34 tests run (0.00s): 34 OK, 0 FAILED, 0 SKIPPED`?

@AlejandroCabeza - It seems a file is missing to be pushed:
image

**Comment by AlejandroCabeza** [comment]
> > Any ideas as to why the `test-ubuntu-latest` fails despite `[Summary] 34 tests run (0.00s): 34 OK, 0 FAILED, 0 SKIPPED`?
> 
> @AlejandroCabeza - It seems a file is missing to be pushed or wrong import statement: image

Oh, okay. Thanks for that, I wouldn't have imagined that's the case. I assummed if it was a missing file error it would be output somewhere else.

**Comment by AlejandroCabeza** [comment]
> LGTM, thanks so much! Same, let's first figure out the segfault before proceeding

I personally would merge it. I've already created a ticket for it so you guys can keep track, and there's nothing in this code (I think) that would change anything in regard to it.

File: tests/waku_lightpush/test_client.nim
Comment (SionoiS) [inline tests/waku_lightpush/test_client.nim:249]: I would remove the `echo`s before merging.

File: tests/waku_lightpush/test_client.nim
Comment (SionoiS) [inline tests/waku_lightpush/test_client.nim:309]: IMO, remove the TODO, add a github issue, or do it.

**Review by SionoiS — CHANGES_REQUESTED** [review]
Thank you!

I would prefer if the segfault was fixed before add a test that will fail.
Also see my comments.

Otherwise, good job.

File: tests/waku_lightpush/test_client.nim
Comment (AlejandroCabeza) [reply tests/waku_lightpush/test_client.nim:309]: Added as an issue, as I don't know how to add more variations.
Ref: https://github.com/waku-org/nwaku/issues/2330

**Comment by AlejandroCabeza** [comment]
Added the `blocked` label, until the segfault has been fixed, as per reviewers suggestion.

**Review by SionoiS — APPROVED** [review]
LGTM thanks!

File: tests/node/test_wakunode_store.nim
Comment (gabrielmer) [inline tests/node/test_wakunode_store.nim:381]: If possible let's move this comment to the correct line, if the formatted doesn't allow it then all good :))

File: tests/waku_lightpush/test_client.nim
Comment (gabrielmer) [reply tests/waku_lightpush/test_client.nim:265]: what's the status with the segfault? Is the PR ready or are we still waiting for this part?

File: tests/node/test_wakunode_store.nim
Comment (AlejandroCabeza) [reply tests/node/test_wakunode_store.nim:381]: That was the autoformatter, yes. It annoyed me too 😂

File: tests/waku_lightpush/test_client.nim
Comment (AlejandroCabeza) [reply tests/waku_lightpush/test_client.nim:265]: That's been fixed. Apparently I forgot to change the comments (I thought I had fixed them), but it's working thanks to a fix by @Ivansete-status

**Review by gabrielmer — APPROVED** [review]
Muchas graciass!

File: tests/resources/pubsub_topics.nim
Comment (AlejandroCabeza) [reply tests/resources/pubsub_topics.nim]: Apparently not. Thanks for pointing that one out :)

**Pull request merged**
